CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPlugin.java

```diff
@@ -146,16 +144,23 @@ public void edit(final GraphWriteMethods graph, final PluginInteraction interact
             }
 
             // Convert world coordinates to camera coordinates.
-            worldLocation.set(x, y, z);
-            worldLocation.subtract(centre);
-            sceneLocation.rotate(worldLocation, rotationMatrix);
-            sceneLocation.setZ(sceneLocation.getZ() - cameraDistance);
+            Vector3f sceneLocation = convertWorldToScene(x, y, z, centre, rotationMatrix, cameraDistance);
+            int rAttr = VisualConcept.VertexAttribute.NODE_RADIUS.get(graph);
+            float r = graph.getFloatValue(rAttr, vxId);
 
             if (sceneLocation.getZ() < 0) {
-                final float horizontalOffset = sceneLocation.getX() / -sceneLocation.getZ();
-                final float verticalOffset = sceneLocation.getY() / -sceneLocation.getZ();
 
-                if (horizontalOffset > left && horizontalOffset < right && verticalOffset < top && verticalOffset > bottom) {
+                final float leftMostPoint = (sceneLocation.getX()-r) / -sceneLocation.getZ();
+                final float rightMostPoint = (sceneLocation.getX()+r) / -sceneLocation.getZ();
+                final float bottomMostPoint = (sceneLocation.getY()-r) / -sceneLocation.getZ();
+                final float topMostPoint = (sceneLocation.getY()+r) / -sceneLocation.getZ();
+                
+                boolean vertexLeftOfBox = rightMostPoint < left;
+                boolean vertexRightOfBox = right < leftMostPoint;
+                boolean vertexBelowBox = topMostPoint < bottom;
+                boolean vertexAboveBox = top < bottomMostPoint;
+                
+                if (!(vertexLeftOfBox || vertexRightOfBox || vertexBelowBox || vertexAboveBox)) {
```

Comment:
Are you able to propagate the `!`?
i.e. `!vertexLeftOfBox && !vertexRightOfBox && !vertexBelowBox && !vertexAboveBox`